pkg/keymanagementprovider/refresh/kubeRefresh.go

```diff
@@ -0,0 +1,182 @@
+			kmp.DeleteKeysFromMap(resource)
+		} else {
+			logger.Error(err, "unable to fetch key management provider")
+		}
+
+		kr.Result = ctrl.Result{}
+
+		return client.IgnoreNotFound(err)
+	}
+
+	lastFetchedTime := metav1.Now()
+	isFetchSuccessful := false
+
+	// get certificate store list to check if certificate store is configured
+	// TODO: remove check in v2.0.0+
+	var certificateStoreList configv1beta1.CertificateStoreList
+	if err := kr.List(ctx, &certificateStoreList); err != nil {
+		logger.Error(err, "unable to list certificate stores")
+		kr.Result = ctrl.Result{}
+		return err
+	}
+
+	if len(certificateStoreList.Items) > 0 {
+		// Note: for backwards compatibility in upgrade scenarios, Ratify will only log a warning statement.
+		logger.Warn("Certificate Store already exists. Key management provider and certificate store should not be configured together. Please migrate to key management provider and delete certificate store.")
+	}
+
+	provider, err := cutils.SpecToKeyManagementProvider(keyManagementProvider.Spec.Parameters.Raw, keyManagementProvider.Spec.Type)
+	if err != nil {
+		writeKMProviderStatus(ctx, kr, &keyManagementProvider, logger, isFetchSuccessful, err.Error(), lastFetchedTime, nil)
+		kr.Request = ctrl.Request{}
+		return err
+	}
+
+	// fetch certificates and store in map
+	certificates, certAttributes, err := provider.GetCertificates(ctx)
+	if err != nil {
+		writeKMProviderStatus(ctx, kr, &keyManagementProvider, logger, isFetchSuccessful, err.Error(), lastFetchedTime, nil)
+		kr.Request = ctrl.Request{}
+		return fmt.Errorf("error fetching certificates in KMProvider %v with %v provider, error: %w", resource, keyManagementProvider.Spec.Type, err)
+	}
+
+	// fetch keys and store in map
+	keys, keyAttributes, err := provider.GetKeys(ctx)
+	if err != nil {
+		writeKMProviderStatus(ctx, kr, &keyManagementProvider, logger, isFetchSuccessful, err.Error(), lastFetchedTime, nil)
+		kr.Request = ctrl.Request{}
+		return fmt.Errorf("error fetching keys in KMProvider %v with %v provider, error: %w", resource, keyManagementProvider.Spec.Type, err)
+	}
+	kmp.SetCertificatesInMap(resource, certificates)
+	kmp.SetKeysInMap(resource, keyManagementProvider.Spec.Type, keys)
+	// merge certificates and keys status into one
+	maps.Copy(keyAttributes, certAttributes)
+	isFetchSuccessful = true
+	emptyErrorString := ""
+	writeKMProviderStatus(ctx, kr, &keyManagementProvider, logger, isFetchSuccessful, emptyErrorString, lastFetchedTime, keyAttributes)
+
+	logger.Infof("%v certificate(s) & %v key(s) fetched for key management provider %v", len(certificates), len(keys), resource)
+
+	// returning empty result and no error to indicate we’ve successfully reconciled this object
+	// will not reconcile again unless resource is recreated
+	if !provider.IsRefreshable() {
+		kr.Request = ctrl.Request{}
+		return nil
+	}
+
+	// if interval is not set, disable refresh
+	if keyManagementProvider.Spec.Interval == "" {
```

Comment:
General comment: I think we should have a good reason to return something as info vs debug log. Ratify logs are already too long :). If you think this will be helpful for users to know immediately, then we can keep it as info. Otherwise, debug is a good option too...

Reply:
On second thought, I don't think it's very useful to the user. It's simply saying that it's working as intended. I'll remove that line. Thanks for the comment. :)